macros/blueprint-proc-macro-playground/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+use gadget_blueprint_proc_macro::{blueprint, job};
+
+#[job(params(n, t), result(_))]
+pub fn keygen(n: u16, t: u8) -> Vec<u8> {
+    let _ = (n, t);
+    vec![0; 33]
+}
+
+blueprint! {
+    metadata: (
+        name: "My Service",
+        version: "1.0.0",
+        description: "My service description",
+    ),
+    jobs: [],
+    registration_hook: Evm("0x4838B106FCe9647Bdf1E7877BF73cE8B0BAD5f97"),
```

Comment:
What happens if this hasn't been deployed yet? Should this not be apart of the pipeline? For example putting an ABI or alloy contract in here instead of an address?

Reply:
Recall what hardhat does with Contract ABIs and deployed bytecode. We could have a similar pattern here. The blueprint generates a blueprint JSON, and similarly there's a deployed blueprint JSON with deployment params/bytecodes/abis.

Reply:
we can embed the contract ABI here in the hook instead of the address, and the `gadget deploy` would handle that and replace it with the deployed contract address when deploying the blueprint on tangle.